contracts/facets/ArchaeologistFacet.sol

```diff
@@ -34,22 +48,88 @@ contract ArchaeologistFacet {
         uint256 withdrawnReward
     );
 
+    function registerArchaeologist(
+        uint256 minimumDiggingFee,
+        uint256 maximumRewrapInterval,
+        uint256 freeBond
+    ) external {
+        // verify that the archaeologist does not already exist
+        LibUtils.revertIfArchProfileExists(msg.sender);
+
+        // create a new archaeologist
+        LibTypes.ArchaeologistProfile memory newArch =
+            LibTypes.ArchaeologistProfile({
+                exists: true,
+                minimumDiggingFee: minimumDiggingFee,
+                maximumRewrapInterval: maximumRewrapInterval,
+                freeBond: freeBond,
+                cursedBond: 0,
+                rewards: 0
+            });
+
+        // transfer SARCO tokens from the archaeologist to this contract, to be
+        // used as their free bond. can be 0.
+        if (freeBond > 0) {
+            s.sarcoToken.transferFrom(msg.sender, address(this), freeBond);
+        }
+
+        // save the new archaeologist into relevant data structures
+        s.archaeologistProfiles[msg.sender] = newArch;
+        s.archaeologistProfileAddresses.push(msg.sender);
+
+        emit RegisterArchaeologist(
+            msg.sender,
+            newArch.minimumDiggingFee,
+            newArch.maximumRewrapInterval,
+            newArch.freeBond
+        );
+    }
+
+    function updateArchaeologist(
+        uint256 minimumDiggingFee,
+        uint256 maximumRewrapInterval,
+        uint256 freeBond
+    ) external {
+        // verify that the archaeologist exists
+        LibUtils.revertIfArchProfileDoesNotExist(msg.sender);
+
+        // create a new archaeologist
+        LibTypes.ArchaeologistProfile storage existingArch = s.archaeologistProfiles[msg.sender];
+        existingArch.minimumDiggingFee = minimumDiggingFee;
+        existingArch.maximumRewrapInterval = maximumRewrapInterval;
+
+        // transfer SARCO tokens from the archaeologist to this contract, to be
+        // used as their free bond. can be 0.
+        if (freeBond > 0) {
+            LibBonds.increaseFreeBond(msg.sender, freeBond);
+            s.sarcoToken.transferFrom(msg.sender, address(this), freeBond);
+        }
+
+        emit UpdateArchaeologist(
+            msg.sender,
+            existingArch.minimumDiggingFee,
+            existingArch.maximumRewrapInterval,
+            existingArch.freeBond
+        );
+    }
```

Comment:
Something occurred to me -- could we not combine register and update into one function? Logically they seem the same, with the only real difference being a "create-if-not-exists" idea. 

On the other hand if `updateArchaeologist` will likely diverge significantly in the future then this comment becomes irrelevant

Reply:
Your comment makes sense. Technically there should be no difference. I have no problem leaving the update function here in case it does diverge like you said.

Reply:
talked about this on standup, but noting here -- these functions will stay separate for now.